syntax = "proto3";
import "common.proto";
import "google/protobuf/empty.proto";
package gsservice;
option cc_generic_services = true;

message EnterGameRequest
{
  uint64 player_id = 1;
  SceneInfo scenes_info = 2;
  uint64 conn_id = 3;
  uint64 gate_node_id = 4;
  uint64 ms_node_id = 5;
}

message EnterGameRespone
{
}

message MsPlayerMessageExtern
{
  uint64 player_id = 1;
}


message MsPlayerMessageRequest
{
  MessageBody msg = 1;
  MsPlayerMessageExtern ex = 2;
}

message MsPlayerMessageRespone
{
  MessageBody msg = 1;
  MsPlayerMessageExtern ex = 2;
}

message RpcClientRequest
{
  bytes request = 1;
  uint64 player_id = 2;
}

message RpcClientResponse
{
  bytes response = 1;
  uint64 conn_id = 2;
}

message DisconnectRequest
{
  uint64 guid = 1;
}

message ConnectRequest
{
  IpPort rpc_client = 1;
  uint32 gate_node_id = 2;
}

message Rg2gTest
{}

service GsService {
  rpc EnterGame (EnterGameRequest) returns (EnterGameRespone);
  rpc PlayerService (MsPlayerMessageRequest) returns (MsPlayerMessageRespone);
  rpc PlayerServiceNoRespone (MsPlayerMessageRequest) returns (google.protobuf.Empty);
  rpc GwPlayerService (RpcClientRequest) returns (RpcClientResponse);
  rpc Disconnect (DisconnectRequest) returns (google.protobuf.Empty); 
  rpc GwConnectGs (ConnectRequest) returns (google.protobuf.Empty); 
  rpc RgEnterRoom (Rg2gTest) returns (google.protobuf.Empty);
}